main.go

```diff
@@ -7,35 +7,134 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/tcnksm/go-gitconfig"
 )
 
 //go:generate go-bindata _templates/...
 
+const defaultVCS = "github.com"
+
+func usage() {
+	fmt.Fprintf(os.Stderr, `Usage of %s:
+	%s new <project name>
+	%s gen -d <model directory -o <output directory>
+	`, os.Args[0], os.Args[0], os.Args[0])
+	os.Exit(1)
+}
+
 func main() {
-	var (
-		modelDir string
-		outDir   string
-	)
 
-	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, `Usage of %s:
-   %s -d <model directory> -o <output directory>
+	if len(os.Args) < 2 {
+		usage()
+	}
+
+	cmd := os.Args[1]
+
+	switch cmd {
+	case "gen":
+		var (
+			modelDir string
+			outDir   string
+		)
+
+		flag := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+		flag.Usage = func() {
+			fmt.Fprintf(os.Stderr, `Usage of %s:
+	%s gen -d <model directory> -o <output directory>
+
+Options:
+`, os.Args[0], os.Args[0])
+			flag.PrintDefaults()
+		}
+
+		flag.StringVar(&modelDir, "d", "", "Model directory")
+		flag.StringVar(&outDir, "o", "", "Output directory")
+
+		flag.Parse(os.Args[2:])
+
+		if modelDir == "" || outDir == "" {
+			flag.Usage()
+			os.Exit(1)
+		}
+		cmdGen(modelDir, outDir)
+
+	case "new":
+		var (
+			vcs      string
+			username string
+		)
+
+		flag := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+		flag.Usage = func() {
+			fmt.Fprintf(os.Stderr, `Usage of %s:
+	%s new <project name>
 
 Options:
 `, os.Args[0], os.Args[0])
-		flag.PrintDefaults()
+			flag.PrintDefaults()
+		}
+
+		flag.StringVar(&vcs, "v", "", "VCS")
+		flag.StringVar(&username, "u", "", "Username")
+
+		if len(os.Args) < 3 {
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		flag.Parse(os.Args[3:])
+
+		if vcs == "" {
+			vcs = defaultVCS
+		}
+
+		if username == "" {
+			var err error
+			username, err = gitconfig.GithubUser()
+
+			if err != nil {
+				username, err = gitconfig.Username()
+				if err != nil {
+					msg := "Cannot find `~/.gitcofig` file.\n" +
+						"Please use -u option"
+					fmt.Println(msg)
+					os.Exit(1)
+				}
+			}
+		}
+
+		project := os.Args[2]
+
+		detail := Detail{vcs, username, project}
+
+		cmdNew(detail)
+
+	default:
+		usage()
 	}
 
-	flag.StringVar(&modelDir, "d", "", "Model directory")
-	flag.StringVar(&outDir, "o", "", "Output directory")
+}
+
+func cmdNew(detail Detail) {
```

Comment:
Argument which is non-primitive type should be passed as pointer. It does not cause copying of entire passed object.